boards/arm/particle_argon_nrf52/particle_argon_nrf52_defconfig

```diff
@@ -0,0 +1,21 @@
+CONFIG_ARM=y
+CONFIG_SOC_FAMILY_NRF=y
+CONFIG_SOC_SERIES_NRF52X=y
+CONFIG_SOC_NRF52840_QIAA=y
+CONFIG_BOARD_PARTICLE_ARGON_NRF52=y
+
+# Enable MPU
+CONFIG_ARM_MPU=y
+
+# enable uart driver
+CONFIG_SERIAL=y
+CONFIG_UART_NRFX=y
+CONFIG_UART_0_NRF_UART=y
```

Comment:
Since you changed the `compatible` property for `uart0` to `"nordic,nrf-uarte"`, you need to update also this entry to `CONFIG_UART_0_NRF_UARTE=y`.

Reply:
Also `CONFIG_UART_1_NRF_UARTE=y` if you want an accessible console.

OTOH, Xenon is picking `UART` rather than `UARTE` for the console, and using `&uart0` for it.  They should be consistent.